### PR TITLE
Packages

### PR DESCRIPTION
**Pull request by gbtami**
[WIP] Heading for https://github.com/gbtami/pychess-variants/issues/492

### CODE REVIEW

File: client/roundCtrl.ts [hunk 1]
```diff
@@ -43,115 +38,73 @@ interface MsgMoreTime {
 }
 
 interface MsgDrawOffer {
```
Comment (Fulmene) [inline client/roundCtrl.ts:40]: These "Msg" types should be in a separate file. Maybe in types.ts or a new file roundType.ts

File: client/roundCtrl.ts [hunk 2]
```diff
@@ -163,6 +116,8 @@ export default class RoundController {
                           // If server received and processed it the first time, it will just ignore it
 
     constructor(el: HTMLElement, model: PyChessModel) {
+        super(el, model);
+        this.materialFunction = updateMaterial;
```
Comment (Fulmene) [inline client/roundCtrl.ts:120]: I don't think an explicit `materialFunction` class member is necessary. Just call `updateMaterial(ctrl)` when appropriate.

File: client/materialView.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+    }
+    return result;
+}
+
+function makeMaterialVNode(ctrl: RoundController, position: 'top'|'bottom', color: cg.Color, content: VNode[], disabled = false): VNode {
+    return h(`div.material.material-${position}.${color}.${ctrl.variant.piece}${disabled ? '.disabled' : ''}`, content);
+}
+
+export function updateMaterial(ctrl: RoundController, vmaterial0?: VNode | HTMLElement, vmaterial1?: VNode | HTMLElement) {
```
Comment (Fulmene) [inline client/materialView.ts:44]: Does this function still need vmaterial parameters anymore?

**Review by Fulmene — CHANGES_REQUESTED** [review]

File: client/materialView.ts (hunk 3 above)
Comment (gbtami) [reply client/materialView.ts]: Good point. Removed.

File: client/materialView.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import * as cg from 'chessgroundx/types';
+import * as util from 'chessgroundx/util';
+
+import { h, VNode } from 'snabbdom';
+
+import { patch } from './document';
+import { calculateGameImbalance, equivalentLetter, MaterialDiff } from './material';
+import { RoundController } from './roundCtrl';
+
+function generateContent(ctrl: RoundController, imbalance: MaterialDiff, color: cg.Color): VNode[] {
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/materialView.ts]: Done.

File: package.json [hunk 5]
```diff
@@ -15,11 +15,11 @@
   },
   "funding": "patreon.com/pychess",
   "dependencies": {
-    "chessgroundx": "^9.1.9",
+    "chessgroundx": "^9.2.0",
```
[the remark this answers is not in the document]
Comment (gbtami) [reply package.json:18]: True. I just used this branch to test 9.2.0
Btw Herroku review apps works again. Anyone can test this branch on https://pychess-pr-943.herokuapp.com/

File: client/lobby.ts [hunk 6]
```diff
@@ -105,7 +104,7 @@ interface Seek {
 
 type CreateMode = 'createGame' | 'playFriend' | 'playAI' | 'createHost';
 
-export class LobbyController {
+export class LobbyController implements IChatController {
     model: PyChessModel;
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/lobby.ts]: Done.

File: client/chess.ts [hunk 7]
```diff
@@ -193,7 +193,7 @@ export class Variant {
 
 }
 
-interface VariantConfig { // TODO explain what each parameter of the variant config means
+export interface IVariant { // TODO explain what each parameter of the variant config means
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/chess.ts]: Done.

File: client/chat.ts [hunk 8]
```diff
@@ -1,19 +1,25 @@
...
-export function chatView(ctrl: RoundController | AnalysisController | TournamentController | LobbyController, chatType: string) { // TODO: instead of | better have some IChatController interface implemented by these classes
+export interface IChatController {
+    anon: boolean;
+    doSend: any;
+    spectator?: boolean;
+    gameId?: string;
+    tournamentId?: string;
+}
+
+export function chatView(ctrl: IChatController, chatType: string) { // TODO: instead of | better have some IChatController interface implemented by these classes
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/chat.ts]: Done.

File: client/cgCtrl.ts [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+            this.ffish = loadedModule;
+            this.ffish.loadVariantConfig(variantsIni);
+            this.notationAsObject = this.notation2ffishjs(this.notation);
+            this.ffishBoard = new this.ffish.Board(this.variant.name, this.fullfen, this.chess960);
+            window.addEventListener('beforeunload', () => this.ffishBoard.delete());
+        });
+    }
+
+    toggleOrientation() {
+        this.flip = !this.flip;
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/cgCtrl.ts]: Done.

File: client/roundCtrl.ts (hunk 1 above)
Comment (gbtami) [reply client/roundCtrl.ts]: Done.

File: client/roundCtrl.ts (hunk 2 above)
Comment (gbtami) [reply client/roundCtrl.ts]: Done.

File: client/cgCtrl.ts [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+import ffishModule from 'ffish-es6';
+
+import { Variant, VARIANTS, notation } from './chess';
+import { boardSettings, IBoardController } from './boardSettings';
+import { PyChessModel } from './types';
+import { variantsIni } from './variantsIni';
+
+export abstract class ChessgroundController implements IBoardController {
+    model: PyChessModel;
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/cgCtrl.ts]: Done.

File: client/sound.ts [hunk 11]
```diff
@@ -42,19 +41,19 @@ class Sounds {
         });
     }
 
-    updateSoundTheme() {
+    updateSoundTheme(assetURL: string) {
```
[the remark this answers is not in the document]
Comment (gbtami) [reply client/sound.ts:44]: Done.

**Review by Fulmene — APPROVED** [review]

**Pull request merged**